src/store/ui/useStudioUIStore.ts

```diff
@@ -19,38 +25,40 @@ export type StudioUIStore = State & Actions
 
 export const useStudioUIStore = create<StudioUIStore>()(
   immer((set) => ({
-    recordings: [],
-    generators: [],
-    scripts: [],
+    recordings: new Map(),
+    generators: new Map(),
+    scripts: new Map(),
```

Comment:
Using `Map` instead of lists here makes it a lot easier to add/remove files. Could be refactored to use `Record` instead if anyone has anything against maps.